Project/src/main/java/app/models/Article.java

```diff
@@ -0,0 +1,141 @@
+package app.models;
+
+import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity
+public class Article
+{
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
+
+    @ManyToMany(cascade = CascadeType.ALL)
+    private List<User> users;
+    private String fileName;
+    private String fileType;
+
+    @ManyToOne
+    private Reviewer reviewer;
+    @OneToOne(cascade = CascadeType.ALL)
+    private Review review;
+    private enum ArticleState {
+        SUBMITTED("Submitted"),
+        IN_REVIEW("In Reivew"),
+        ACCEPTED("Accepted"),
+        REJECTED("Rejected");
+
+        private final String text;
+
+        private ArticleState(String text) {
+            this.text = text;
+        }
+
+        @Override
+        public String toString() {
+            return text;
+        }
+    }
+    private ArticleState state;
+
+    @Lob
+    private byte[] data;
+
+    public Article(Reviewer reviewer) {
+        this.reviewer = reviewer;
+        this.users = new ArrayList<>();
+
+        if (reviewer != null) {
+            state = ArticleState.IN_REVIEW;
+        }
+        else {  // reviewer == null
+            state = ArticleState.SUBMITTED;
```

Comment:
Shouldn't State.SUBMITTED be for when the review has been submitted and not when the file has been submitted for review? I.e. In review (when submitter submits file) -> Submitted (when reviewer adds review) -> Accepted | Rejected (when editor makes decision based on review)

Reply:
no, the submitted state is when the file gets uploaded

Reply:
When does it switch to in review then? And what's between in review and a decision?

Reply:
Should switch to in review when a reviewer is assigned an article